src/clr-core/icon-shapes/icon.element.scss

```diff
@@ -366,19 +312,13 @@ svg {
 :host(.has-badge--error[inverse]),
 :host(.has-badge--danger.is-inverse),
 :host(.has-badge--error.is-inverse) {
-  .clr-i-badge {
-    @include setIconColor(
-      #{'var(--clr-icon-color-inverse-danger, var(--clr-color-danger-500,' + $clr-color-danger-500 + '))'}
-    );
-  }
+  --badge-color: #{$clr-color-danger-500};
 }
 
 // DEPRECATED IN 3.0: is-highlight, is-inverse
 :host([inverse].is-highlight),
 :host(.is-highlight.is-inverse) {
-  @include setIconColor(
-    #{'var(--clr-icon-color-inverse-info, var(--clr-color-action-400,' + $clr-color-action-400 + '))'}
-  );
+  --color: $clr-color-action-400;
```

Comment:
I'm really against how we are relying on SASS variables throughout all of this code. 

In either case, this line is missing the `#{}`